File: ios/Sources/SimInfoPlugin/SimInfoPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SimInfoPlugin)
public class SimInfoPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "SimInfoPlugin"
    public let jsName = "SimInfo"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = SimInfo()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/vinteum/cap/siminfo/SimInfoPlugin.java
package com.vinteum.cap.siminfo;

import com.getcapacitor.PluginCall;
import com.getcapacitor.JSObject;

public class SimInfoPlugin {

    private SimInfo implementation = new SimInfo();
    
    public void callEcho(PluginCall call) {
        String value = call.getString("value");
        if (value != null) {            
            implementation.echo(call);
        } else {
            call.reject("Value is missing");
        }
    }
}
